### DISCUSSION TITLE
Skip currently refreshing assets with Auto-Materialization

### DISCUSSION THREAD

**Discussion by quickcoffee** [dagster-io/dagster#19214] [Q&A] [answered]
Hey, we want to use [auto-materialization](https://docs.dagster.io/concepts/assets/asset-auto-execution) to refresh dbt root assets (most upstream assets) with a cron schedule, but struggle with parallel runs of the same asset (eg. two separate runs triggered by fresh upstream data). Is there any skip rule we can add to prevent launching new runs for an asset if it is still being refreshed?
skip_on_backfill_in_progress sounds promising, but only works for backfills?

**Comment by rexledesma** [comment, marked as answer]
In Dagster `1.6.12`, we shipped a `AutoMaterializeRule.skip_on_run_in_progress()`: https://github.com/dagster-io/dagster/pull/20498.

This will skip sending a run from AMP if there is already a run in progress for the particular asset.

**Comment by OwenKephart** [comment]
This is now the out-of-the-box behavior of the new `AutomationCondition.eager()`.

If you are building a [custom Automation Condition](https://docs.dagster.io/concepts/automation/declarative-automation/customizing-automation-conditions), then you can always add `AutomationCondition.<something>() & ~AutomationCondition.in_progress()` to ensure that the asset is never requested if it currently has an in-progress run.
